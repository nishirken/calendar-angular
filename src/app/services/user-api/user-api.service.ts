import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { url } from '../helpers';
import { User } from './user-api.interfaces';

@Injectable({
  providedIn: 'root',
})
export class UserApiService {
  constructor(private http: HttpClient) {}

  getUser(): Observable<User> {
    return this.http.get<User>(url('/user'), { withCredentials: true });
  }
}
