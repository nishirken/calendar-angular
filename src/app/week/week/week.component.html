<div [attr.data-test-id]="testIds.root" class="root">
    <app-sidebar></app-sidebar>
</div>
