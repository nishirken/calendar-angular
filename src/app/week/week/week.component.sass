:host
    display: block
    height: 100%

.root
    height: 100%