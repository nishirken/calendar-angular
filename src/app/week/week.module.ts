import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WeekComponent } from './week/week.component';
import { UserApiService } from '../services/user-api/user-api.service';
import { SidebarModule } from '../sidebar/sidebar.module';

@NgModule({
  declarations: [WeekComponent],
  imports: [CommonModule, SidebarModule],
  providers: [UserApiService],
})
export class WeekModule {}
