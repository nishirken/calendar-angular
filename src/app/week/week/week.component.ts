import { Component, OnDestroy, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { weekTestIds } from 'calendar-test-ids';
import { Subscription } from 'rxjs';
import { DateTimeService } from '../../services/date-time/date-time.service';
import { UserApiService } from '../../services/user-api/user-api.service';

@Component({
  selector: 'app-week',
  templateUrl: './week.component.html',
  styleUrls: ['./week.component.sass'],
})
export class WeekComponent implements OnInit, OnDestroy {
  private routeParamMapSub: Subscription | null = null;

  constructor(
    private readonly userApiService: UserApiService,
    private readonly route: ActivatedRoute,
    private readonly dateTimeService: DateTimeService
  ) {}

  testIds = weekTestIds;

  ngOnInit(): void {
    this.routeParamMapSub = this.route.paramMap.subscribe((params) =>
      this.dateTimeService.weakParamMap$.next(params)
    );
    this.userApiService.getUser().subscribe(console.log);
  }

  ngOnDestroy(): void {
    this.routeParamMapSub?.unsubscribe();
  }
}
