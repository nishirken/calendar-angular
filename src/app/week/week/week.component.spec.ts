import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { UserApiService } from '../../services/user-api/user-api.service';
import { UserApiService as UserApiServiceStub } from '../../services/user-api/__mocks__/user-api.service';
import { WeekComponent } from './week.component';

describe('WeekComponent', () => {
  let component: WeekComponent;
  let fixture: ComponentFixture<WeekComponent>;
  let userApiServiceStub = new UserApiServiceStub();

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ WeekComponent ],
      providers: [ { provide: UserApiService, useValue: userApiServiceStub, } ],
      schemas: [ CUSTOM_ELEMENTS_SCHEMA ],
    })
    .compileComponents();

    fixture = TestBed.createComponent(WeekComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('Renders', () => {
    expect(fixture).toMatchSnapshot();
  });
});
