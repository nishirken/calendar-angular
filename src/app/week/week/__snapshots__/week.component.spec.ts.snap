// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekComponent Renders 1`] = `
<app-week
  __ngContext__="0"
  dateTimeService={[Function DateTimeService]}
  route={[Function Object]}
  routeParamMapSub={[Function SafeSubscriber]}
  testIds={[Function Object]}
  userApiService={[Function UserApiService]}
>
  <div
    class="root"
    data-test-id="week"
  >
    <app-sidebar />
  </div>
</app-week>
`;
